gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -234,6 +236,14 @@ public final String getGdchApiAudience() {
     return gdchApiAudience;
   }
 
+  /**
+   * Gets the ApiKey that should be used for authentication. If an empty string was provided it will
+   * return null
+   */
+  public final String getApiKey() {
+    return (apiKey == null || apiKey.isEmpty()) ? null : apiKey;
```

Comment:
If a customer explicitly sets apiKey to empty, I think it is a mistake and we should error out instead of silently ignoring it, hence we should just return whatever is set directly.

Reply:
I removed from now.  Waiting to see how we perform validation in the end as to where we will check for nulls